nixos/modules/installer/tools/nixos-generate-config.pl

```diff
@@ -577,14 +587,64 @@ sub multiLineList {
   # Set your time zone.
   # time.timeZone = "Europe/Amsterdam";
 
-  # List packages installed in system profile. To search by name, run:
-  # \$ nix-env -qaP | grep wget
+  # List of packages installed in system profile (by attribute name).
+  #
   # environment.systemPackages = with pkgs; [
   #   wget vim
   # ];
 
+  # To search packages by name, run:
+  #
+  # \$ nix-env -qaP '.*wget.*'
+  #
+  # To search by name and description you can do something like
+  #
+  # \$ nix-env -qaP --description | grep -i distributed | grep -i storage
+  #
+  # The first field in the outputs of those commands in the attribute name
+  # you can use in `environment.systemPackages` above.
+  #
+  # Note, however, that Nixpkgs and by consequence NixOS make unfree
+  # packages unavailable by default. That is,
+  #
+  # \$ nix-env -qaP '.*skype.*'
+  #
+  # will provide no useful results and
+  #
+  # environment.systemPackages = [ pkgs.skype ];
+  # (You might need to merge this line with the above mention of
+  # environment.systemPackages as you can't assign the same attribute
+  # multiple times in a single nix attribute set.)
+  #
+  # will fail to evaluate.
+  #
+  # To enable unfree packages in this configuration.nix
+  # uncomment the following
+  #
+  # nixpkgs.config.allowUnfree = true;
+  #
+  # To make unfree packages available in nix-env, nix-build,
+  # nix-shell, etc you can either run them with NIXPKGS_ALLOW_UNFREE=1
+  # environment variable set, e.g.
+  #
+  # \$ NIXPKGS_ALLOW_UNFREE=1 nix-env -qaP '.*skype.*'
+  #
+  # or you can add
+  #
+  #     { allowUnfree = true; }
+  #
+  # into your ~/.config/nixpkgs/config.nix file.
+  #
+  # Finally, note that if you don't use the NixOS channels, you can
+  # still use nix-env by running it directly from the nixpkgs tree like so
+  #
+  # \$ nix-env -f . -qaP '.*wget.*'
+
```

Comment:
:+1:

Reply:
Given that the whole point of not showing unfree software by default is to not promote the use of unfree software, adding detailed instructions to the default configuration.nix is questionable IMHO.

Reply:
@edolstra Well, it is half of the point — the other half is that we have too many different evaluation contexts to ever be sure that `allowUnfree` is not silently true in one of them.

(I do _personally_ agree with the point you make, although I value the fail-safe part more and I am not sure if people stopping opening issues like #17126 is worth documenting this in more places)